demos/matrix_free/stokes.py.rst

```diff
@@ -0,0 +1,50 @@
+==================
+ Stokes Equations
+==================
+Again, they are what they are.  This is the driven cavity example.
+We have several options files showing how to do some field-splitting
+and such for this.::
+
+  from firedrake import *
+
+  N = 128
+
+  M = UnitSquareMesh(N, N)
+
+  V = VectorFunctionSpace(M, "CG", 2)
+  W = FunctionSpace(M, "CG", 1)
+  Z = V * W
+
+  u, p = TrialFunctions(Z)
+  v, q = TestFunctions(Z)
+
+  a = (inner(grad(u), grad(v)) -
+       p * div(v) +
+       div(u) * q + p*q)*dx
+
+  F = Function(V)
+  G = Function(W)
+
+  L = inner(F, v) * dx + G*q*dx
+
+  bcs = [DirichletBC(Z.sub(0), Constant((1, 0)), (4,)),
+         DirichletBC(Z.sub(0), Constant((0, 0)), (1, 2, 3))]
+
+  up = Function(Z)
+
+  nullspace = MixedVectorSpaceBasis(
+      Z, [Z.sub(0), VectorSpaceBasis(constant=True)])
```

Comment:
Yes, pressure is only defined up to a constant.

On Mon, Aug 15, 2016 at 8:03 AM, Lawrence Mitchell <notifications@github.com
> wrote:

> In demos/matrix_free/stokes.py.rst
> <https://github.com/firedrakeproject/firedrake/pull/853#discussion_r74757664>
> :
>
> > +  a = (inner(grad(u), grad(v)) -
> > +       p * div(v) +
> > +       div(u) * q + p*q)*dx
> > +
> > +  F = Function(V)
> > +  G = Function(W)
> > +
> > +  L = inner(F, v) * dx + G*q*dx
> > +
> > +  bcs = [DirichletBC(Z.sub(0), Constant((1, 0)), (4,)),
> > +         DirichletBC(Z.sub(0), Constant((0, 0)), (1, 2, 3))]
> > +
> > +  up = Function(Z)
> > +
> > +  nullspace = MixedVectorSpaceBasis(
> > +      Z, [Z.sub(0), VectorSpaceBasis(constant=True)])
>
> Is the nullspace because there are no pressure boundary conditions?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/firedrakeproject/firedrake/pull/853/files/ff5ec4e0eabb94576ac69c9c82932eaa50b2d14e#r74757664>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADl6LwT6lu7VO6nBheYT7aQYeYJlTPNuks5qgGOFgaJpZM4JiV-g>
> .
>